numpy/typing/tests/data/reveal/array_constructors.pyi

```diff
@@ -20,6 +20,9 @@ i8: np.int64
 A: npt.NDArray[np.float64]
 B: SubClass[np.float64]
 C: list[int]
+D: SubClass[np.float64 | np.int64]
```

Comment:
will this also work for custom type that implements `__array__`, but isn't a subtype of `ndarray`?